### DISCUSSION TITLE
Alter Template Helper editor to avoid user confusion.

### DISCUSSION THREAD

**Discussion by Didgeridrew** [home-assistant/frontend#20832] [Other feature requests] [answered]
Since its introduction, there have been regular posts on the forums and Discord from users who are confused that their helpers are always "unavailable" or by error messages they receive in the template renderer within the editor.  The vast majority of the time this is due to the user not understanding that the "State template" field is for templates only, not YAML configuration.

[Example 1](https://community.home-assistant.io/t/help-template-sensor-not-working/718032)
[Example 2](https://community.home-assistant.io/t/extracting-temperature-from-climate-entity-in-template-sensor-ui/731040)
[Example 3](https://community.home-assistant.io/t/template-sensor-that-updates-every-x-minutes-or-hours/690592)

The UI of the editor needs to make it more obvious/explicit that the "State template" field is only for Jinja templating and that "advanced" options like triggers, actions, attributes, and availability are not supported. 

If there is no current plan to expand the Template Helper to allow the "advanced" options, we also need to address the [Template Docs: UI Configuration](https://www.home-assistant.io/integrations/template/#ui-configuration) to also make these limitations explicit and include an example screenshot.

**Comment by karwosts** [comment]
Perhaps something like this? I'm thinking maybe it's sufficient to just look for the string `template:`, `sensor:` , or `state:` in the user input, or is that ever a valid syntax that would be a false positive? (though it is just a warning so that wouldn't obstruct anything)



![image](https://github.com/home-assistant/frontend/assets/32912880/9ea941db-6a65-4107-a414-e2596e4fbb8f)


If you think the docs are missing something, feel free to PR some additional language there that you think would be helpful.

**Reply by Didgeridrew** [reply]
I think that's good, as long as it also catches `binary_sensor:`, `platform:`, and  `value_template:`... since ChatGPT seems to prefer the Legacy style template sensors config, so we're seeing a lot of the last two.

**Reply by Didgeridrew** [reply]
It looks like the warning may be needed for template fields elsewhere... 

https://community.home-assistant.io/t/send-phone-notification-based-on-email-subject/742638

**Reply by karwosts** [reply, marked as answer]
Completed @ https://github.com/home-assistant/frontend/pull/21545

**Comment by karwosts** [comment]
@silamon - probably doesn't make sense to close this as nothing has been implemented yet 😕
